sims/benchmark-simuPOP.py

```diff
@@ -0,0 +1,330 @@
+
+sim.setRNG(seed=args.seed)
+random.seed(args.seed)
+
+logfile = fileopt(args.logfile, "w")
+
+logfile.write("Options:\n")
+logfile.write(str(args)+"\n")
+logfile.write(time.strftime('%X %x %Z')+"\n")
+logfile.write("----------\n")
+logfile.flush()
+
+# define defaults for benchmark
+npops = 1
+rloci = args.rho / (args.popsize * 4 * args.recomb_rate)
+uloci = args.theta / (args.popsize * 4 * args.mut_rate)
+assert rloci == uloci
+args.nloci = int(rloci)
+args.sel_mut_rate = args.mut_rate * args.pdel
+
+
+# increase spacing between loci as we go along the chromosome
+rel_positions=[0.0 for k in range(args.nloci-1)]
+for k in range(1,args.nloci-1):
+    rel_positions[k] = rel_positions[k-1] + random.expovariate(1)*(k**2)
+pos_fac=args.length/(rel_positions[-1] + random.expovariate(1)*(k**2))
+locus_position=[x*pos_fac for x in rel_positions] + [args.length]
+
+# initially polymorphic alleles
+init_freqs=[[k/100,1-k/100,0,0] for k in range(1,11)]
+locus_classes=[min(len(init_freqs)-1,math.floor(random.expovariate(1))) for k in range(args.nloci)]
+init_classes=[list(filter(lambda k: locus_classes[k]==x,range(args.nloci))) for x in range(len(init_freqs))]
+
+logfile.write("Locus positions:\n")
+logfile.write(str(locus_position)+"\n")
+logfile.write("----------\n")
+logfile.flush()
+
+###
+# modified from http://simupop.sourceforge.net/manual_svn/build/userGuide_ch5_sec9.html
+
+class GammaDistributedFitness:
+    def __init__(self, shape, scale):
+        # mean is shape/scale
+        self.coefMap = {}
+        # these are called alpha/beta by `random.gammavariate` but are
+        # implemented as shape/scale paramterization
+        #
+        self.alpha = shape
+        self.beta = scale
+    def __call__(self, loc, alleles):
+        # because s is assigned for each locus, we need to make sure the
+        # same s is used for fitness of genotypes 01 (1-s) and 11 (1-2s)
+        # at each locus
+        if loc in self.coefMap:
+            s = self.coefMap[loc]
+        else:
+            s = random.gammavariate(self.alpha, self.beta)
+            self.coefMap[loc] = s
+        # print(str(loc)+":"+str(alleles)+"\n")
+        # needn't return fitness for alleles=(0,0) as simupop knows that's 1
+        if 0 in alleles:
+            return max(0.0, 1. - s)
+        else:
+            return max(0.0, 1. - 2.*s)
+
+time_dict = {'prepping': None,
+             'simulating': None,
+             'finalizing': None}
+
+start = time.time()
+
+init_geno=[sim.InitGenotype(freq=init_freqs[k],loci=init_classes[k])
+           for k in range(len(init_freqs))]
+
+pop = sim.Population(
+        size=[args.popsize]*npops, 
+        loci=[args.nloci], 
+        lociPos=locus_position,
+        infoFields=['ind_id','fitness','migrate_to'])
+
+id_tagger = sim.IdTagger()
+id_tagger.apply(pop)
+
+# set up recomb collector
+first_gen = pop.indInfo("ind_id")
+init_ts = msprime.simulate(2*len(first_gen),
+                           length=max(locus_position))
```

Comment:
Need to leave this for now, as `RecombCollector` is not modified (as `ArgRecorder` was by https://github.com/ashander/ftprime/pull/40) to use an empty ts by default